src/backend/base/langflow/api/v1/flows.py

```diff
@@ -256,20 +257,26 @@ async def read_flows(
 async def _read_flow(
     session: AsyncSession,
     flow_id: UUID,
-    user_id: UUID,
+    current_user: CurrentActiveUser,
     settings_service: SettingsService,
 ):
     """Read a flow."""
     auth_settings = settings_service.auth_settings
+    access_service = get_access_service()
     stmt = select(Flow).where(Flow.id == flow_id)
+
     if auth_settings.AUTO_LOGIN:
         # If auto login is enable user_id can be current_user.id or None
         # so write an OR
         stmt = stmt.where(
-            (Flow.user_id == user_id) | (Flow.user_id == None)  # noqa: E711
+            (Flow.user_id == current_user.id) | (Flow.user_id == None)  # noqa: E711
         )
-    return (await session.exec(stmt)).first()
+    flow = (await session.exec(stmt)).first()
+    
+    if flow and access_service.enforcer.enforce(current_user, flow, "view"):
```

Comment:
This would represent the actual access checking with Casbin while keeping the current db architecture.

I am not 100% sure what would be the best/most performant way of querying the flows. Especially when loading a list of flows (e..g in the get("/") read_flows endpoint here). 
One option would be to load all flows form the db and filter the access based on casbin. However, this seems to be rather inefficient especially in terms of memory usage.

Alternatively, we could pre-filter the flows with the SQL statements as it is done right now (e.g. `stmt = select(Flow).where(Flow.user_id == current_user.id)`). Doing this would be more efficient but it would also mix the access model filter approach -> why use casbin in the first place then?

Any input would be appreciated